gen/calc.py

```diff
@@ -575,6 +575,30 @@ def fz_err(msg):
             validate_int_in_range(port, 1, 65535)
 
 
+def calculate_scarce_resources(enable_scarce_resources):
```

Comment:
Do we even need all this validation in here if you have a separate `validate` function below?